external/avro/src/main/scala/org/apache/spark/sql/avro/CatalystDataToAvro.scala

```diff
@@ -19,19 +19,24 @@ package org.apache.spark.sql.avro
 
 import java.io.ByteArrayOutputStream
 
+import org.apache.avro.Schema
 import org.apache.avro.generic.GenericDatumWriter
 import org.apache.avro.io.{BinaryEncoder, EncoderFactory}
 
 import org.apache.spark.sql.catalyst.expressions.{Expression, UnaryExpression}
 import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, ExprCode}
 import org.apache.spark.sql.types.{BinaryType, DataType}
 
-case class CatalystDataToAvro(child: Expression) extends UnaryExpression {
+case class CatalystDataToAvro(
+    child: Expression,
+    jsonFormatSchema: Option[String]) extends UnaryExpression {
```

Comment:
Can we have a default value `None`?
```scala
-   jsonFormatSchema: Option[String]) extends UnaryExpression {
+   jsonFormatSchema: Option[String] = None) extends UnaryExpression {
```

Reply:
Here I am trying to avoid parameter with a default value. The result is quite different with/without a specified schema. 
Also, this is consistent with `CatalystDataToAvro`.

Reply:
unless the default value is used a lot in tests, I don't think we should add default value in internal classes. We should force the caller side to specify the parameter when they instantiate the internal class.

Reply:
Got it, @gengliangwang and @cloud-fan .